node/bft/src/gateway.rs

```diff
@@ -955,6 +968,23 @@ impl<N: Network> Gateway<N> {
             }
         }
     }
+
+    /// Processes a message received from the network.
+    async fn process_message_inner(&self, peer_addr: SocketAddr, message: Event<N>) -> io::Result<()> {
```

Comment:
```suggestion
    async fn process_message_inner(&self, peer_addr: SocketAddr, message: Event<N>) {
```
since the body is infallible; otherwise `clippy` will lint us about it someday

Reply:
Because the number of failures needs to be counted, Result is returned.
https://github.com/AleoNet/snarkOS/blob/09aa62b4b877c5e96689ba2e0c77d5fd1c69ae7c/node/tcp/src/protocols/reading.rs#L144-L147
The current modification has a problem: putting process_message_inner in the tokio::spawn task will not return the processing result, so it is impossible to count the failures.
```Rust
/// Processes a message received from the network.
async fn process_message(&self, peer_addr: SocketAddr, message: Self::Message) -> io::Result<()> {
    if matches!(message, Event::BlockRequest(_) | Event::BlockResponse(_)) {
        let self_ = self.clone();
        // Handle BlockRequest and BlockResponse messages in a separate task to not block the
        // inbound queue.
        tokio::spawn(async move {
            self_.process_message_inner(peer_addr, message).await;
        });
    } else {
        self.process_message_inner(peer_addr, message).await;
    }
    Ok(())
}
```

Reply:
Note that due to the way we've been calling `process_message` all this time hasn't utilized this feature; also, if `process_message_inner` is infallible, it wouldn't change anything.

Failure-handling is performed in `inbound` IIRC, and the lower-level network stack doesn't need to be involved.

Reply:
Do you mean we can ignore the failure statistics for now?
I am worried that when we discover abnormal statistical data in the future, we will have to spend time to investigate.

Reply:
What I'm saying is that we've never used the low-level failure statistics; `process_message` needs to return a `Result` due to the lower-level API, but it has always been infallible in snarkOS.

Reply:
Ok, I get it.